src/test/java/com/faforever/client/player/FollowedPlayerJoinedGameNotifierTest.java

```diff
@@ -0,0 +1,110 @@
+
+public class FollowedPlayerJoinedGameNotifierTest {
+  private FollowedPlayerJoinedGameNotifier instance;
+  @Mock
+  private NotificationService notificationService;
+  @Mock
+  private EventBus eventBus;
+  @Mock
+  private I18n i18n;
+  @Mock
+  private JoinGameHelper joinGameHelper;
+  @Mock
+  private Preferences preferences;
+  @Mock
+  private NotificationsPrefs notification;
+  @Mock
+  private AudioService audioService;
+  @Mock
+  private TaskService taskService;
+
+  @Before
+  public void setUp() throws Exception {
+    MockitoAnnotations.initMocks(this);
+
+    instance = new FollowedPlayerJoinedGameNotifier(notificationService, i18n, eventBus, joinGameHelper, audioService, taskService);
+
+    instance.postConstruct();
+    verify(eventBus).register(instance);
+  }
+
+  @Test
+  public void testSubscribeAnnotations() {
+    assertThat(ReflectionUtils.findMethod(instance.getClass(), "onFollowedJoinedGame", FollowedPlayerJoinedGameEvent.class),
+        hasAnnotation(Subscribe.class));
+  }
+
+  @Test
+  public void onFollowedPlayerJoinedGame() throws Exception {
+    Game game = GameBuilder.create().defaultValues().title("My Game").get();
+    Player player = PlayerBuilder.create("junit").id(1).game(game).get();
+
+    when(i18n.get("followed.joinedGameNotification.title", "junit", "My Game")).thenReturn("following junit");
+    when(i18n.get("followed.joinedGameNotification.action")).thenReturn("Click to cancel");
+
+    instance.onFollowedJoinedGame(new FollowedPlayerJoinedGameEvent(player, game));
+
+    ArgumentCaptor<TransientNotification> captor = ArgumentCaptor.forClass(TransientNotification.class);
+    verify(notificationService).addNotification(captor.capture());
+
+    TransientNotification notification = captor.getValue();
+    assertThat(notification.getTitle(), is("following junit"));
+    assertThat(notification.getText(), is("Click to cancel"));
+    assertThat(notification.getImage(), notNullValue());
+
+    NoCatch.noCatch(() -> Thread.sleep(3500));//TODO shorten
+
+    verify(joinGameHelper).join(game);
+  }
+
+  @Test
+  public void onFollowedPlayerJoinedGameCANCELLED() throws Exception {
+    Game game = GameBuilder.create().defaultValues().title("My Game").get();
+    Player player = PlayerBuilder.create("junit").id(1).game(game).get();
+
+    when(i18n.get("followed.joinedGameNotification.title", "junit", "My Game")).thenReturn("following junit");
+    when(i18n.get("followed.joinedGameNotification.action")).thenReturn("Click to cancel");
+
+    instance.onFollowedJoinedGame(new FollowedPlayerJoinedGameEvent(player, game));
+
+    ArgumentCaptor<TransientNotification> captor = ArgumentCaptor.forClass(TransientNotification.class);
+    verify(notificationService).addNotification(captor.capture());
+
+    TransientNotification notification = captor.getValue();
+    notification.getActionCallback().call(null);
+
+    NoCatch.noCatch(() -> Thread.sleep(3500));//TODO shorten
```

Comment:
really bad, using mocked taskService I could just verify for call/capture the task given as parameter and execute it?